### ISSUE TITLE
Add sorting option to reference fields

### ISSUE THREAD

**Issue by bjornwang** [sanity-io/sanity#3143]
**Is your feature request related to a problem? Please describe.**
Sometimes reference fields are used for small, well-defined sets of configuration values. In these scenarios it would be great to be able to sort them.

For example, in this case (sorry - retina image), the document type has a length field used for sorting which would be great to use for the reference selector too -- the shortest period at the top of the list. (The list contains predefined values used in integration with a legacy system.)

![image](https://user-images.githubusercontent.com/5180175/153246997-eacfcd39-9938-42f0-b1c3-1b7014ca8c7f.png)

**Describe the solution you'd like**
As we have optional filters for references today, also add a sort order option.

**Describe alternatives you've considered**
Prefixing the values with numbers or alphabetic characters to affect the order.

**Comment by jonathanquaade** [comment]
Any update on this? This feature would be life-changing

**Comment by miked-known** [comment]
Agreed, this is a sorely needed feature particularly for reference arrays with mixed types... Any ETA on this?

**Comment by sebbean** [comment]
i, personally, would really enjoy this feature immensely

**Comment by qruz-hq** [comment]
Would be cool 🙏

**Comment by ErlendKH** [comment]
Can anyone confirm that it still works? No sorting for me - I'm using Sanity 3.54.0.

**Comment by tomahg** [comment]
Still works! Using Sanity 3.61.0.
